src/test/java/com/google/api/gax/grpc/UnaryCallableTest.java

```diff
@@ -458,25 +462,106 @@ public Integer extractPageSize(Integer payload) {
     }
   }
 
-  private class PagedListResponse extends PagedListResponseImpl<Integer, List<Integer>, Integer> {
-    public PagedListResponse(
+  private class ListIntegersPagedResponse implements PagedListResponse<Integer> {
+
+    private final PagedListResponseContext<Integer, List<Integer>, Integer> context;
+    private final ListIntegersPage page;
+
+    public ListIntegersPagedResponse(
         UnaryCallable<Integer, List<Integer>> callable,
         PagedListDescriptor<Integer, List<Integer>, Integer> pageDescriptor,
         Integer request,
-        CallContext context) {
-      super(callable, pageDescriptor, request, context);
+        CallContext callContext) {
+      this.context = new PagedListResponseContext<>(callable, pageDescriptor, request, callContext);
+      this.page = new ListIntegersPage(this.context);
+    }
+
+    @Override
+    public Iterable<Integer> iterateAll() {
+      return context.iterateAll();
+    }
+
+    @Override
+    public ListIntegersPage getPage() {
+      return page;
+    }
+
+    @Override
+    public Iterable<ListIntegersPage> iteratePages() {
+      return new Iterable<ListIntegersPage>() {
+        @Override
+        public Iterator<ListIntegersPage> iterator() {
+          return new PageContext.PageIterator<>(
+              new PageFetcher<ListIntegersPage>() {
+                @Override
+                public ListIntegersPage getNextPage(ListIntegersPage currentPage) {
+                  return currentPage.getNextPage();
+                }
+              },
+              page);
+        }
+      };
+    }
+
+    @Override
+    public String getNextPageToken() {
+      return context.getNextPageToken();
+    }
+
+    @Override
+    public FixedSizeCollection<Integer> expandToFixedSizeCollection(int collectionSize) {
+      return context.expandToFixedSizeCollection(collectionSize);
+    }
+
+    @Override
+    public Iterable<FixedSizeCollection<Integer>> iterateFixedSizeCollections(int collectionSize) {
+      return context.iterateFixedSizeCollections(collectionSize);
+    }
+  }
+
+  private class ListIntegersPage implements Page<Integer> {
+
+    private final PageContext<Integer, List<Integer>, Integer> context;
+
+    public ListIntegersPage(PageContext<Integer, List<Integer>, Integer> context) {
+      this.context = context;
+    }
```

Comment:
Similar to above, if a `Page` implementation just ends up forwarding everything to the context, maybe the better pattern would be to make an `AbstractPage`. Then, you would have:

```
class ListIntegersPage extends AbstractPage<Integer> {}
```